cli/.eslintrc.js

```diff
@@ -20,6 +20,7 @@ module.exports = {
     "total-functions",
   ],
   rules: {
+    "functional/prefer-immutable-types": ["error", { "enforcement": "ReadonlyShallow" }],
```

Comment:
This PR gets us up to `ReadonlyShallow`. I might go for `ReadonlyDeep` in a subsequent PR. I probably won't bother with truly `Immutable`.

Reply:
I went all the way to immutable